packages/@winglang/sdk/src/target-tf-aws/function.ts

```diff
@@ -284,15 +287,29 @@ export class Function extends cloud.Function implements IAwsFunction {
 
     const bundle = createBundle(this.entrypoint, externalLibraries);
 
-    // would prefer to create TerraformAsset in the constructor, but using a CDKTF token for
-    // the "path" argument isn't supported
-    const asset = new TerraformAsset(this, "Asset", {
-      path: bundle.directory,
-      type: AssetType.ARCHIVE,
-    });
+    if (bundle.directory.includes(" ")) {
+      //since terraformAsset can't handle file paths that contain spaces- we create the bundle zip manually.
+      const zip = new AdmZip();
+      zip.addLocalFolder(bundle.directory);
+
+      const folderPath = join(App.of(this).outdir, "assets");
+      const filename = `${this.node.addr}.zip`;
+      ensureDirSync(folderPath);
+      zip.writeZip(join(folderPath, filename));
+
+      this.assetPath = join("assets", filename);
+    } else {
+      // would prefer to create TerraformAsset in the constructor, but using a CDKTF token for
+      // the "path" argument isn't supported
+      const asset = new TerraformAsset(this, "Asset", {
+        path: bundle.directory,
+        type: AssetType.ARCHIVE,
+      });
+
+      this.assetPath = asset.path;
+    }
 
     this.bundleHash = bundle.hash;
-    this.assetPath = asset.path;
   }
```

Comment:
Is there a reason we need to still use TerraformAsset? Can we bundle the zip manually in all cases?

Reply:
The manual zip makes the process longer in more than 25%- so the benchmark tests are failing (this is why I split it), I think we can consider remove it in the future